### PR TITLE
chore(docs): Migrating from Apollo Guide

### LINKED ISSUE #28: Docs: Migrating from Apollo Guide
- `ApolloClient` ➡️`GraphQLClient`
- `<Query />` ➡️ `useQuery`
- `<Mutation />` ➡️`useMutation`
- `client.query` ➡️`useManualQuery`
- SSR `getDataFromTree` ➡️`getIntialState`

### PR DESCRIPTION
**Pull request by bmullan91**
### What does this PR do?

Provides a complete guide on how to migrate from `apollo` ➡️`graphql-hooks`. Rendered output: https://github.com/nearform/graphql-hooks/blob/70c9242f7817188611d31ba539f1bda41df23725/README.md

#### TODOs

- [x] `<Mutation />` ➡️`useMutation()` guide

### Related issues

Fixes #28 

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+-  uri: '/graphql',
+-  cache: new InMemoryCache()
+- })
++ const client = new GraphQLClient({
++   url: '/graphql',
++   cache: memCache()
++ })
+```
+
+Alot of the options you'd pass to `ApolloClient` are the same as `GraphQLClient`:
```
Comment (jackdclark) [inline README.md]:
```suggestion
A lot of the options you'd pass to `ApolloClient` are the same as `GraphQLClient`:
```

File: README.md [hunk 2]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+        return <div>{data}</div>
+-      }}
+-    </Query>
+-  )
+}
+```
+
+### Query Component Props
+
+Alot of options can be carried over as-is, or have direct replacement:
```
Comment (jackdclark) [inline README.md]:
```suggestion
A lot of options can be carried over as-is, or have direct replacements:
```

File: README.md [hunk 3]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+- `uri` ➡️ `url`
+- `fetchOptions`
+- `onError` - the function signature is slighly different
+- `headers`
+- `fetch`
+- `cache`
+
+### ApolloProvider ➡️ ClientContext.Provider
+
+```diff
```
Comment (jackdclark) [inline README.md]: Didn't realise you could do this, it's awesome 👏

File: README.md [hunk 4]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+```
+
+### Query Component Props
+
+Alot of options can be carried over as-is, or have direct replacement:
+
+- `query` ➡️ `useQuery(query)`: No need to wrap the query in `gql`
+- `variables` ➡️ `useQuery(query, { variables })`
+- `ssr` ➡️ `useQuery(query, { ssr })`
+- **Fetch Policies**: See [#75](https://github.com/nearform/graphql-hooks/issues/75) for a more info
```
Comment (jackdclark) [inline README.md]:
```suggestion
- **Fetch Policies**: See [#75](https://github.com/nearform/graphql-hooks/issues/75) for more info
```

File: README.md [hunk 5]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+
+```diff
+- <Query query={gql`...`}>
+-  {(props) => {}}
+- </Query>
++ const state = useQuery(query)
+```
+
+- `props.loading` ➡️ `state.loading`
+- `props.error`: The error prop from `useQuery` is Boolean the details of the error can be found in either:
```
Comment (jackdclark) [inline README.md]:
```suggestion
- `props.error`: The error value from `useQuery` is Boolean, the details of the error can be found in either:
```

File: README.md [hunk 6]
```diff
@@ -453,17 +457,181 @@ export default function PostList() {
...
+- `optimisticResponse`
+- `refetchQueries`
+- `awaitRefetchQueries`
+- `context`
+
+## Mutation Component Render Props
+
+- `data` ➡️ `const [mutateFn, { data }] = useMutation`
+- `loading` ➡️ `const [mutateFn, { loading }] = useMutation`
+- `error` ➡️ `const [mutateFn, { error }] = useMutation`
```
Comment (jackdclark) [inline README.md]: Do we need a similar description to `useQuery` here? Our `error` is a boolean

File: README.md [hunk 7]
```diff
@@ -453,17 +457,191 @@ export default function PostList() {
...
+        return <div>{data}</div>
+-      }}
+-    </Query>
+-  )
+}
+```
+
+### Query Component Props
+
+A lot of options can be carried over as-is, or have direct replacement:
```
Comment (jackdclark) [inline README.md]:
```suggestion
A lot of options can be carried over as-is, or have direct replacements:
```

File: README.md [hunk 8]
```diff
@@ -453,17 +457,191 @@ export default function PostList() {
...
+  - `cache-only`: Not supported
+  - `no-cache` ➡️ `useQuery(QUERY, { useCache: false })`
+
+**Not yet supported**
+
+- `errorPolicy`: Any error will set the `error` to be truthy. See [useQuery](#useQuery) for more details.
+- `pollInterval`
+- `notifyOnNetworkStatusChange`
+- `skip`
+- `onCompleted`: Similar ability if using `useManualQery`
```
Comment (Joezo) [inline README.md]:
```suggestion
- `onCompleted`: Similar ability if using `useManualQuery`
```

File: README.md [hunk 9]
```diff
@@ -453,17 +457,191 @@ export default function PostList() {
...
+- `onError`: Similar ability if using `useManualQery`
```
Comment (Joezo) [inline README.md]:
```suggestion
- `onError`: Similar ability if using `useManualQuery`
```

**Pull request merged**